baseplate/sidecars/live_data_watcher.py

```diff
@@ -128,25 +73,67 @@ def on_change(self, data: Optional[bytes], _znode_stat: ZnodeStat) -> None:
             if self.owner and self.group:
                 os.fchown(tmpfile.fileno(), self.owner, self.group)
             os.fchmod(tmpfile.fileno(), self.mode)
-            try:
-                json_data = json.loads(data.decode("UTF-8"))
-            except json.decoder.JSONDecodeError:
-                # If JSON fails to decode, still write the bytes data since
-                # we don't necessarily know if the the contents of the znode
-                # had to be valid JSON anyways.
-                tmpfile.write(data)
-            else:
-                # If no exceptions, we have valid JSON, and can parse accordingly.
-                data = NodeWatcher.get_data_to_write(json_data, data)
-                if data is None:
-                    logger.warning(
-                        "No data written to destination file. Something is likely misconfigured."
-                    )
-                    return
-                tmpfile.write(data)
+            tmpfile.write(data)
         os.rename(self.dest + ".tmp", self.dest)
 
 
+def _parse_loader_type(data: Optional[bytes]) -> LoaderType:
+    try:
+        json_data = json.loads(data.decode("UTF-8"))
+    except json.decoder.JSONDecodeError:
+        logger.debug("Data is not valid JSON. Loading as RAW")
+        return LoaderType.RAW
+
+    try:
+        loader_type = json_data["live_data_watcher_load_type"]
+    except (json.decoder.JSONDecodeError, KeyError, AttributeError, TypeError):
+        logger.debug("Data is not a JSON object like {\"live_data_watcher_load_type\":\"load_type\"}. Loading as RAW")
+        return LoaderType.RAW
+
+    return LoaderType(loader_type)
+
+
+def _load_from_s3(data: bytes) -> Optional[bytes]:
+    loader_config = json.loads(data.decode("UTF-8"))
```

Comment:
:no_bell: I think it's likely a good idea that we are representing this data as JSON and not ini, but using ini would allow us to use the built in config parsing spec. It might be a good idea to add a comment here about why we are using this json and not ini.

Reply:
🔕  Do we use ini config formats for anything other than loading ini files?

Reply:
Honestly I didn't even consider not using JSON for this, writing out ini from scratch in code is a PITA in my experience. I don't have much of a reason other than that, but wowsers I don't wanna change it either

Reply:
> no_bell Do we use ini config formats for anything other than loading ini files?

All application and sidecar configuration is in ini, and the [baseplate config library](https://github.com/reddit/baseplate.py/blob/develop/baseplate/lib/config.py) allows you to define a spec for your configuration that would take care of the validation below. I'm not aware of anyone using this outside of Baseplate application and sidecar configuration though,

Reply:
> Honestly I didn't even consider not using JSON for this, writing out ini from scratch in code is a PITA in my experience. I don't have much of a reason other than that, but wowsers I don't wanna change it either

To be clear I don't should use ini. It might be worth pointing out that we are doing some of the things that the baseplate config library does. To me the most compelling reason to use json, and not ini is it's simpler. It allows someone who is writing data to zk to verify their data a lot easier.